File: Examples/test-suite/perl5/template_typedef_runme.pl
use strict;
use warnings;
use Test::More tests => 7;
BEGIN { use_ok('template_typedef') }
require_ok('template_typedef');

my $d = template_typedef::make_Identity_float();
like(ref($d), qr/^template_typedef::/);
my $c = template_typedef::make_Identity_reald();
like(ref($c), qr/^template_typedef::/);

my $e = template_typedef::make_Multiplies_float_float_float_float($d, $d);
like(ref($e), qr/^template_typedef::/);

my $f = template_typedef::make_Multiplies_reald_reald_reald_reald($c, $c);
like(ref($f), qr/^template_typedef::/);

my $g = template_typedef::make_Multiplies_float_float_reald_reald($d, $c);
like(ref($g), qr/^template_typedef::/);

File: Examples/test-suite/python/template_typedef_runme.py
from template_typedef import *

d = make_Identity_float()
c = make_Identity_reald()


try:
    a = d.this
    a = c.this
except:
    raise RuntimeError

try:
    e = make_Multiplies_float_float_float_float(d, d)
    a = e.this
except:
    raise RuntimeError("{} is not an instance".format(e))

try:
    f = make_Multiplies_reald_reald_reald_reald(c, c)
    a = f.this
except:
    raise RuntimeError("{} is not an instance".format(f))

try:
    g = make_Multiplies_float_float_reald_reald(d, c)
    a = g.this
except:
    raise RuntimeError("{} is not an instance".format(g))


# the old large format
if not SWIG_TypeQuery("vfncs::ArithUnaryFunction<vfncs::arith_traits<float,double>::argument_type,vfncs::arith_traits<float,double >::result_type > *"):
    raise RuntimeError


# the reduced format
if not SWIG_TypeQuery("vfncs::ArithUnaryFunction<double,double> *"):
    raise RuntimeError

# this is a bad name
if SWIG_TypeQuery("vfncs::ArithUnaryFunction<double,doublex> *"):
    raise RuntimeError
